pkg/util/version_history.go

```diff
@@ -75,7 +99,7 @@ func logVersionHistoryToFile(versionHistoryFilePath, agentVersion string, timest
 		return
 	}
 
-	err = ioutil.WriteFile(versionHistoryFilePath, file, 0644)
```

Comment:
What's the difference between `ioutil.WriteFile` and `os.WriteFile`?

Reply:
`ioutil.WriteFile` behaves same as `os.WriteFile` and it is [deprecated since go 1.16](https://go.dev/doc/go1.16#ioutil).
https://github.com/golang/go/blob/346b18ee9d15410ab08dd583787c64dbed0666d2/src/io/ioutil/ioutil.go#L46 

I refer to `release-branch.go1.17` because I think [this repo uses go 1.17](https://github.com/DataDog/datadog-agent/blob/c186fc840c518bcaff01076a34977229f55e761d/go.mod#L3).